nerfstudio/data/pixel_samplers.py

```diff
@@ -40,7 +40,7 @@ def collate_image_dataset_batch(batch: Dict, num_rays_per_batch: int, keep_full_
     if "mask" in batch:
         nonzero_indices = torch.nonzero(batch["mask"][..., 0], as_tuple=False)
```

Comment:
Does it by any chance work if you instead do
`nonzero_indices = torch.nonzero(batch["mask"][..., 0].to(device), as_tuple=False)`
This puts it on the gpu earlier which might be a little faster

Reply:
that seems to work too, but FYI 'device' in this case seems to be cpu (sorry, had a typo in the original PR description)